File: redirector.rb
require 'nokogiri'
require 'open-uri'
require 'json'
require 'redis'
require 'time'
require 'mongo'
$client = Mongo::Client.new([ "localhost" ], :database => "redirects", :max_pool_size => 100, :wait_queue_timeout => 60, :connect_timeout => 30, :socket_timeout => 30)
Mongo::Logger.logger.level = Logger::FATAL
$redis = Redis.new
latest = "https://permanent-redirect.xyz/pages/1515462547"
click_count = $client[:redirects].find.sort({click_count: -1}).first["click_count"] rescue 0
data = Nokogiri.parse(open(latest).read());false
while true
  if data.search("a").length == 2
    sleep(20)
    data = Nokogiri.parse(open(latest).read());false
  else
    latest = "https://permanent-redirect.xyz/pages/"+data.search("a")[0].attributes["href"].value.split("/").last
    data = Nokogiri.parse(open(latest).read());false
    latest_id = latest.split("/").last
    timestamp = Time.parse(data.search("p")[1].text.split("created at ").last.split(".").first)
    click_count += 1
    page_visitors = data.search("img").collect{|x| x.attributes["alt"].text}.join.to_i
    current_status = {time: Time.now, page_visitors: page_visitors, clicked_link: latest, latest_id: latest_id, timestamp: timestamp, click_count: click_count}
    $client[:redirects].insert_one(current_status)
    $redis.set("redirect_current_status", current_status.to_json)
  end
end

File: redirector.py
from selenium import webdriver
import numpy as np
import json
driver = webdriver.Chrome("./chromedriver")
latest = "https://permanent-redirect.xyz/pages/1515462547"
driver.get(latest)
click_count = 0
while True:
  latest = driver.find_elements_by_tag_name("a")[0].get_attribute("href")
  driver.find_elements_by_tag_name("a")[0].click()
  latest_id = str.split(str(latest), "/")[-1]
  timestamp = str.split(str(driver.find_elements_by_tag_name("p")[1].text), "created at ")[-1]
  click_count += 1
  present_on_page = int(str.join("", [str(el.get_attribute("alt")) for el in driver.find_elements_by_tag_name("img")]))
  print json.dumps({"page_visitors": present_on_page, "clicked_link": latest, "latest_id": latest_id, "timestamp": timestamp, "click_count": click_count})
